### PR TITLE
Add export to video dataset

### PR DESCRIPTION
**Pull request by horatiualmasan**
## What has changed and why?

Add export functionality to the video dataset.

## How has it been tested?

New test

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

* **New Features**
  * Added video dataset export functionality with support for YouTube-VIS instance segmentation format:
    * New export module and class: VideoDatasetExport (export/export_video_dataset.py) with method to_youtube_vis_instance_segmentation.
    * New top-level helper function to_youtube_vis_instance_segmentation(session, samples, output_json).
    * VideoDataset gained an export(...) method to create a VideoDatasetExport for a dataset/query.
    * Exports include categories, videos, annotations (instance segmentation RLE, bboxes, areas, frame references, object track ids), and related video/frame metadata.
* **Tests**
  * Added test lightly_studio/tests/export/test_export_video_dataset.py covering export to YouTube-VIS, validating JSON structure for categories, videos, and annotations across frames and object tracks.
* **Dependencies**
  * Updated dependency pin in lightly_studio/pyproject.toml: replaced `labelformat>=0.1.12` with a git-based reference `labelformat @ git+https://github.com/lightly-ai/labelformat.git@4928eee1ee89798f674878e68a6eb4cc897cf357`.
* **Changelog**
  * CHANGELOG.md was not updated for this internal change.

### CODE REVIEW

File: lightly_studio/src/lightly_studio/export/export_video_dataset.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class VideoDatasetExport:
+    """Provides methods to export a video dataset or a subset of it."""
+
+    def __init__(self, session: Session, samples: Iterable[VideoSample]):
+        """Initialize the export with a database session and the video samples to export."""
+        self.session = session
+        self.samples = samples
+
+    def to_youtube_vis_instance_segmentation(self, output_json: PathLike | None = None) -> None:
```
Comment (LeonardoRosaa) [inline lightly_studio/src/lightly_studio/export/export_video_dataset.py]: We could define:

```
to_youtube_vis_instance_segmentation(self, output_json: PathLike = DEFAULT_EXPORT_FILENAME)
```

Then we can remove the condition below.

File: lightly_studio/src/lightly_studio/export/export_video_dataset.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        )
+
+
+def to_youtube_vis_instance_segmentation(
+    session: Session,
+    samples: Iterable[VideoSample],
+    output_json: Path,
+) -> None:
+    """Export video instance segmentation tracks to a YouTube-VIS JSON file."""
+    adapter = youtube_vis_label_input.LightlyStudioYouTubeVISInstanceSegmentationTrackInput(
```
Comment (LeonardoRosaa) [inline lightly_studio/src/lightly_studio/export/export_video_dataset.py]: I'd rename it to `label_input`.

File: lightly_studio/tests/export/test_export_video_dataset.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                    annotation_label_id=label.annotation_label_id,
+                    annotation_type=AnnotationType.INSTANCE_SEGMENTATION,
+                    x=0,
+                    y=1,
+                    width=1,
+                    height=1,
+                    # Row-wise RLE for a 2x3 mask with a single 1 at (row=0, col=1):
+                    # row-major pixels: [0,1,0,0,0,0] -> counts [1,1,4]
+                    segmentation_mask=[1, 1, 4],
+                    object_track_id=object_track_id,
```
Comment (LeonardoRosaa) [inline lightly_studio/tests/export/test_export_video_dataset.py:51]: I’d suggest adding a test with `object_track_id` set to `None`, if that’s a possible use case.

File: lightly_studio/tests/export/test_export_video_dataset.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                "id": 1,
+                "file_names": ["video_001.mp4/00000.jpg", "video_001.mp4/00001.jpg"],
+                "width": 3,
+                "height": 2,
+                "length": 2,
+            }
+        ]
+        assert yvis["annotations"] == [
+            {
+                "id": 99,
```
Comment (LeonardoRosaa) [inline lightly_studio/tests/export/test_export_video_dataset.py:86]: The reader should understand the details to understand that this ID is the track number. I'd add a comment.

File: lightly_studio/src/lightly_studio/export/export_video_dataset.py (hunk 2 above)
Comment (horatiualmasan) [reply lightly_studio/src/lightly_studio/export/export_video_dataset.py]: renamed to `export_input`

File: lightly_studio/tests/export/test_export_video_dataset.py (hunk 3 above)
Comment (horatiualmasan) [reply lightly_studio/tests/export/test_export_video_dataset.py:51]: Added a new annotation with object_track_id set to None. It is ignored in the export.
